app/views/documents/index.js.erb

```diff
@@ -0,0 +1,42 @@
+$("tbody").replaceWith("<tbody><%= j render partial: "documents/document", collection: @documents.select {|document|
+                    document.category == @category} %></tbody>");
+
+<%# I'd like to add the code between the comments to a different file, so that this file is organized better %>
+function date_asc(){
+  $("#sort-date").replaceWith("<%= j link_to "Date", documents_path(sort: "created_at", direction: "ASC", category: @category.to_s), class:"created_at", id:"sort-date", remote: true %>");
+};
+
+function date_desc(){
+  $("#sort-date").replaceWith("<%= j link_to "Date", documents_path(sort: "created_at", direction: "DESC", category: @category.to_s), class:"created_at desc", id:"sort-date", remote: true %>");
+};
+
+function name_asc(){
+  $("#sort-name").replaceWith("<%= j link_to "Name", documents_path(sort: "filename", direction: "ASC", category: @category.to_s), class:"filename", id:"sort-name", remote: true %>");
+};
+
+function name_desc(){
+  $("#sort-name").replaceWith("<%= j link_to "Name", documents_path(sort: "filename", direction: "DESC", category: @category.to_s), class:"filename desc", id:"sort-name", remote: true %>");
+};
+
+function sort_class(){
+  <% if  @sorting_params.include?("created_at") %> 
+    if (document.querySelector("#sort-date").classList.contains("desc")){
+      date_asc();
+    } else {
+      date_desc();
+    };
+    name_asc();
+  <% elsif @sorting_params.include?("filename") %>
+    if (document.querySelector("#sort-name").classList.contains("desc")){
+      name_asc();
+    } else {
+      name_desc();
+    };
+    date_asc();
+  <% else %>
+    name_desc();
+  <% end %>
+}
+<%# I need to find a way to require .js.erb in this file so all I'll need to do is execute the function%>
+
+sort_class()
```

Comment:
Put this method call before all the function definitions, as this helps the reader understand what we actually do here. A bit like a summary. JS functions are hoisted before the actual code is run, so you can call them from anywhere. So the ideal writing style would be to define methods by "abstraction level", which allows the reader to not having to read code that they don't care about.
e.g.
```javascript
sortClass();

function sortClass() { ... }
function dateDesc() { ... }
...
```

The naming of `#sort_class()` should be improved (and camelCased, it's the JS convention), because this function does not sort classes, it does not sort anything either. The method name should actually describe what is done, otherwise we force the reader to read everything to understand what is actually done.